examples/barotropic_tide.jl

```diff
@@ -0,0 +1,285 @@
+                                      forcing = (u = tidal_forcing,))
+
+# We initialize the model with the tidal flow and a linear stratification.
+
+uᵢ(x, y, z) = U_tidal
+
+Nᵢ² = 2e-4  # [s⁻²] initial buoyancy frequency / stratification
+bᵢ(x, y, z) = Nᵢ² * z
+
+set!(model, u=uᵢ, b=bᵢ)
+
+# Now let's built a `Simulation`.
+
+Δt = 3minutes
+stop_time = 4days
+
+simulation = Simulation(model; Δt, stop_time)
+
+# We add a callback to print a message about how the simulation is going,
+
+using Printf
+
+wall_clock = Ref(time_ns())
+
+function print_progress(sim)
+
+    elapsed = 1e-9 * (time_ns() - wall_clock[])
+
+    msg = @sprintf("iteration: %d, time: %s, wall time: %s, max|w|: %6.3e, m s⁻¹\n",
+                   iteration(sim), prettytime(sim), prettytime(elapsed),
+                   maximum(abs, sim.model.velocities.w))
+
+    wall_clock[] = time_ns()
+
+    @info msg
+
+    return nothing
+end
+
+simulation.callbacks[:print_progress] = Callback(print_progress, IterationInterval(200))
+nothing #hide
+
+# ## Diagnostics/Output
+
+# Add some diagnostics. Instead of ``u`` we save the deviation of ``u`` from its instantaneous
+# domain average, ``u' = u - (L_x H)^{-1} \int u \, \mathrm{d}x \mathrm{d}z``. We also save
+# the stratification ``N^2 = \partial_z b``.
+
+b = model.tracers.b
+u, v, w = model.velocities
+
+U = Field(Average(u))
+
+u′ = u - U
+
+N² = ∂z(b)
+
+filename = "barotropic_tide"
+save_fields_interval = 30minutes
+
+simulation.output_writers[:fields] = JLD2OutputWriter(model, (; u, u′, w, b, N²);
+                                                      filename,
+                                                      schedule = TimeInterval(save_fields_interval),
+                                                      overwrite_existing = true)
+
+# We are ready -- let's run!
+
+run!(simulation)
+
+# ## Load output
+
+# First, we load the saved valocity and stratification output as `FieldTimeSeries`es.
+
+saved_output_filename = filename * ".jld2"
+
+u′_t = FieldTimeSeries(saved_output_filename, "u′")
+ w_t = FieldTimeSeries(saved_output_filename, "w")
+N²_t = FieldTimeSeries(saved_output_filename, "N²")
+
+times = u′_t.times
+nothing #hide
+
+# We retrieve each field's coordinates and convert from meters to kilometers.
+
+xu,  yu,  zu  = nodes(u′_t[1]) ./ 1e3
+xw,  yw,  zw  = nodes(w_t[1])  ./ 1e3
+xN², yN², zN² = nodes(N²_t[1]) ./ 1e3
+nothing #hide
+
+# ## Visualize
+
+# Now we can visualize our resutls! We use `CairoMakie` here. On a system with OpenGL
+# `using GLMakie` is more convenient as figures will be displayed on the screen.
+
+using CairoMakie
+
+# First, a utility to mask the region that is within the immersed boundary with `NaN`s and
+# also select the ``x``-``z`` interior slices of our fields.
+
+using Oceananigans.ImmersedBoundaries: mask_immersed_field!
+
+function mask_and_get_interior(φ_t, n; value=NaN)
+    mask_immersed_field!(φ_t[n], value)
+    return interior(φ_t[n], :, 1, :)
+end
+nothing #hide
```

Comment:
We shouldn't have to use this in an example... seems liek something is missing from the API?

Reply:
Yes!! #3092 tries to sort this out!

Reply:
The example is a little early then.. convert to draft?

Reply:
This is not crucial -- I can delete it.

Reply:
Just don't want to encourage hacking 😂

Reply:
Yes, I agree -- I don't like that either!

Reply:
But it's not hacking! It's clean and honest.
But it's using non-user facing functionality which is *definitely* not ideal!